### PR TITLE
[SAI Test] Add L3 Neighbor Test

### PR DESCRIPTION
**Pull request by zhoudongxu**
add l3 neighbor test

### CODE REVIEW

**Review by Gfrom2016 — APPROVED** [review]

**Comment by rlhui** [comment]
@zhoudongxu  - please update PR title name.

**Comment by zhoudongxu** [comment]
> @zhoudongxu - please update PR title name.

update

File: test/sai_test/sai_neighbor.py [hunk 1]
```diff
@@ -0,0 +1,840 @@
...
+    """
+
+    def setUp(self):
+        """
+        Test the basic setup process.
+        """
+        T0TestBase.setUp(self)
+          
+        self.dev_port1 = self.dev_port_list[1]
+        self.ipv4_addr = "01.1.1.10"
```
Comment (richardyu-ms) [inline test/sai_test/sai_neighbor.py:17]: 10.1.1.10

File: test/sai_test/sai_neighbor.py [hunk 2]
```diff
@@ -0,0 +1,840 @@
...
+            pass
+
+    def tearDown(self):
+        sai_thrift_remove_neighbor_entry(self.client, self.nbr_entry_v6)
+        super().tearDown()
+
+class RemoveAddNeighborTestIPV4(T0TestBase):
+    """
+    Verifies if IPv4 host route is not created according to
+    SAI_NEIGHBOR_ENTRY_ATTR_NO_HOST_ROUTE attribute value
```
Comment (richardyu-ms) [inline test/sai_test/sai_neighbor.py:225]: add the correct test objective in the description for the test class

File: test/sai_test/sai_neighbor.py [hunk 3]
```diff
@@ -0,0 +1,840 @@
...
+        self.assertEqual(self.status(), SAI_STATUS_SUCCESS)
+
+        net_route = sai_thrift_route_entry_t(vr_id=self.default_vrf, destination=sai_ipprefix(self.ipv4_addr+'/32'))
+        sai_thrift_create_route_entry(self.client, net_route, next_hop_id=self.lag1_rif)       
+        self.assertEqual(self.status(), SAI_STATUS_SUCCESS)
+
+    
+    def RemoveAddNeighborTestV4(self):
+        '''
+        Verifies
```
Comment (richardyu-ms) [inline test/sai_test/sai_neighbor.py:254]: add test steps in this method or in the runTest method

File: test/sai_test/sai_neighbor.py [hunk 4]
```diff
@@ -0,0 +1,840 @@
...
+        self.nbr_entry_v4 = sai_thrift_neighbor_entry_t(
+                rif_id=self.lag1_rif,
+                ip_address=sai_ipaddress(self.ipv4_addr))
+        status = sai_thrift_create_neighbor_entry(
+                self.client,
+                self.nbr_entry_v4,
+                dst_mac_address=self.mac_addr)              
+        self.assertEqual(self.status(), SAI_STATUS_SUCCESS)
+
+        net_route = sai_thrift_route_entry_t(vr_id=self.default_vrf, destination=sai_ipprefix(self.ipv4_addr+'/32'))
```
Comment (richardyu-ms) [inline test/sai_test/sai_neighbor.py:247]: please check if the new created items are removed in the teardown method

File: test/sai_test/sai_neighbor.py [hunk 5]
```diff
@@ -0,0 +1,840 @@
+from sai_test_base import T0TestBase
+from sai_utils import *
+
```
Comment (richardyu-ms) [inline test/sai_test/sai_neighbor.py:3]: the file name should be sai_neighbor_test.py

File: test/sai_test/config/port_configer.py [hunk 6]
```diff
@@ -442,6 +443,30 @@ def get_mtu(self):
         '''
         return int(self.config.get('mtu'))
 
+    def get_cpu_port_queue(self):
```
Comment (richardyu-ms) [inline test/sai_test/config/port_configer.py:446]: add comments for this method
move the assignement method to the helper. explicit change the test's attributes outside of the configer(in the help method). then we can reuse this method just for getting the attribute but without changing the test object

File: test/sai_test/config/route_configer.py [hunk 7]
```diff
@@ -21,26 +21,30 @@
 
 from sai_thrift.sai_adapter import *
 from sai_utils import *  # pylint: disable=wildcard-import; lgtm[py/polluting-import]
+import pdb
```
Comment (richardyu-ms) [inline test/sai_test/config/route_configer.py:24]: pdb?

File: test/sai_test/config/port_configer.py (hunk 6 above)
Comment (zhoudongxu) [reply test/sai_test/config/port_configer.py:453]: we can get cpu queue by self.cpuqueue{num}

**Pull request merged**